src/main/java/uk/gov/hmcts/darts/courthouse/service/CourthouseServiceImpl.java

```diff
@@ -48,4 +52,43 @@ public Courthouse addCourtHouse(uk.gov.hmcts.darts.courthouse.model.Courthouse c
         Courthouse mappedEntity = this.mapper.mapToEntity(courthouse);
         return repository.saveAndFlush(mappedEntity);
     }
+
+
+    /**
+     * retrieves the courtroom from the database.
+     *
+     * @param courthouseCode Optional parameter. If it is not provided, then name will be used by itself.
+     * @param courthouseName Name of the courthouse to search for.
+     * @return the found courtroom
+     * @throws CourthouseNameNotFoundException when the courthouse isn't found
+     * @throws CourthouseCodeNotMatchException when the courtroom is found, but it has a different code that expected.
+     */
+    @Override
+    public Courthouse retrieveCourtHouse(Integer courthouseCode, String courthouseName)
```

Comment:
would it be possible to put the update in a different location , feels like retrieve method doesn't imply there is an update.

Reply:
have renamed